src/main/java/io/prometheus/cloudwatch/CloudWatchCollector.java

```diff
@@ -613,6 +620,33 @@ private void scrape(List<MetricFamilySamples> mfs) throws CloneNotSupportedExcep
         for (Map.Entry<String, ArrayList<MetricFamilySamples.Sample>> entry : extendedSamples.entrySet()) {
           mfs.add(new MetricFamilySamples(baseName + "_" + safeName(toSnakeCase(entry.getKey())), Type.GAUGE, help(rule, unit, entry.getKey()), entry.getValue()));
         }
+        
+        // Add the "aws_resource_info" metric for existing tag mappings
+        for (ResourceTagMapping resourceTagMapping : resourceTagMappings) {
+          if (!publishedResourceInfo.contains(resourceTagMapping.getResourceARN())) {
+            List<String> labelNames = new ArrayList<String>();
+            List<String> labelValues = new ArrayList<String>();
+            labelNames.add("job");
+            labelValues.add(jobName);
+            labelNames.add("instance");
+            labelValues.add("");
+            labelNames.add("arn");
+            labelValues.add(resourceTagMapping.getResourceARN());
+            labelNames.add(safeName(toSnakeCase(rule.awsTagSelect.resourceIdDimension)));
+            labelValues.add(extractResourceIdFromArn(resourceTagMapping.getResourceARN()));
```

Comment:
Can this be done via iteration somehow instead of going line by line?

Reply:
Maybe it can, but I currently don't see a way to that without it to be overkill and complex. it's currently simple and easy to understand.